### PR TITLE
[Improvement](multi catalog) The interface of external Splitter. WIP

### PR DESCRIPTION
**Pull request by Jibing-Li**
This is PR introduce splitter interface external table. The splitter interface contain one method `getSplits`, which is used by QueryScanProvider to get the external file split. For Hive/Iceberg/TVF, a split is a file block. For ES, it is a shard.
This PR also move the getSplits logic in FileScanProviderIf to the new Splitter interface.
In the future, we may unify internal table as well.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/Split.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+package org.apache.doris.planner;
+
+import lombok.Data;
+import org.apache.hadoop.fs.Path;
+
+@Data
+public class Split {
+
+    private Path path;
```
[the remark this answers is not in the document]
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/planner/Split.java:26]: Added FileSplit

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/TVFSplitter.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.doris.thrift.TBrokerFileStatus;
+
+import com.google.common.collect.Lists;
+import org.apache.hadoop.fs.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+public class TVFSplitter implements Splitter {
```
[the remark this answers is not in the document]
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/planner/external/TVFSplitter.java:34]: Will fix this in next pr for TVF

**Comment by Jibing-Li** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**